code/modules/reagents/Chemistry-Goon-420BlazeIt.dm

```diff
@@ -143,4 +143,329 @@ datum/reagent/crank/addiction_act_stage4(var/mob/living/M as mob)
+		step(M, pick(cardinal))
+	M.Jitter(15)
+	M.Dizzy(15)
+	if(prob(40))
+		M.emote(pick("twitch","drool","moan"))
+	..()
+	return
+/datum/reagent/methamphetamine/addiction_act_stage4(var/mob/living/carbon/human/M as mob)
+	if(M.canmove && !istype(M.loc, /turf/space))
+		step(M, pick(cardinal))
+		step(M, pick(cardinal))
+		step(M, pick(cardinal))
+		step(M, pick(cardinal))
+		step(M, pick(cardinal))
+		step(M, pick(cardinal))
+		step(M, pick(cardinal))
+		step(M, pick(cardinal))
+	M.Jitter(20)
+	M.Dizzy(20)
+	M.adjustToxLoss(5)
+	if(prob(50))
+		M.emote(pick("twitch","drool","moan"))
+	..()
+	return
+
+/datum/chemical_reaction/methamphetamine
+	name = "methamphetamine"
+	id = "methamphetamine"
+	result = "methamphetamine"
+	required_reagents = list("ephedrine" = 1, "iodine" = 1, "phosphorous" = 1, "hydrogen" = 1)
+	result_amount = 4
+	required_temp = 374
+
+/datum/chemical_reaction/methamphetamine_two
+	name = "methamphetamine_two"
+	id = "methamphetamine_two"
+	result = "methamphetamine"
+	required_reagents = list("muriatic_acid" = 1, "caustic_soda" = 1, "hydrogen_chloride" = 1)
+	result_amount = 3
+	required_temp = 374
+
+/datum/reagent/muriatic_acid
+	name = "Muriatic Acid"
+	id = "muriatic_acid"
+	description = "Fuck me, we needed those cooks."
+	reagent_state = LIQUID
+	color = "#60A584" // rgb: 96, 165, 132
+
+/datum/reagent/caustic_soda
+	name = "Caustic Soda"
+	id = "caustic_soda"
+	description = "Fuck me, we needed those cooks."
+	reagent_state = LIQUID
+	color = "#60A584" // rgb: 96, 165, 132
+
+/datum/reagent/hydrogen_chloride
+	name = "Hydrogen Chloride"
+	id = "hydrogen_chloride"
+	description = "Fuck me, we needed those cooks."
+	reagent_state = LIQUID
+	color = "#60A584" // rgb: 96, 165, 132
+
+/datum/chemical_reaction/muriatic_acid
+	name = "muriatic_acid"
+	id = "muriatic_acid"
+	result = "muriatic_acid"
+	required_reagents = list("mutadone" = 1, "sacid" = 1)
+	result_amount = 2
+	required_temp = 500
+
+/datum/chemical_reaction/caustic_soda
+	name = "caustic_soda"
+	id = "caustic_soda"
+	result = "caustic_soda"
+	required_reagents = list("sacid" = 1, "cola" = 1)
+	result_amount = 2
+	required_temp = 500
+
+/datum/chemical_reaction/hydrogen_chloride
+	name = "hydrogen_chloride"
+	id = "hydrogen_chloride"
+	result = "hydrogen_chloride"
+	required_reagents = list("hydrogen" = 1, "chlorine" = 1)
+	result_amount = 2
+	required_temp = 500
+
+/datum/chemical_reaction/saltpetre
+	name = "saltpetre"
+	id = "saltpetre"
+	result = "saltpetre"
+	required_reagents = list("potassium" = 1, "nitrogen" = 1, "oxygen" = 3)
+	result_amount = 3
+
+/datum/reagent/saltpetre
+	name = "Saltpetre"
+	id = "saltpetre"
+	description = "Volatile."
+	reagent_state = LIQUID
+	color = "#60A584" // rgb: 96, 165, 132
+
+/datum/reagent/bath_salts
+	name = "Bath Salts"
+	id = "bath_salts"
+	description = "Makes you nearly impervious to stuns and grants a stamina regeneration buff, but you'll be a nearly uncontrollable tramp-bearded raving lunatic."
```

Comment:
Correct me if I'm wrong, but I think the apostrophe in you'll is treated as the end of the desc here.